### PR TITLE
Document the BrowserType class in its own page and document the top level module properties on the landing page

### PR DESCRIPTION
**Pull request by ankur22**
Closes: https://github.com/grafana/xk6-browser/issues/486

Now that we've changed how we import and launch a browser process (https://github.com/grafana/xk6-browser/pull/462), we also need to move the documentation out of the landing page and into its own class -- `BrowserType`. Eventually we will need to do the same for the other browsers we support.

### CODE REVIEW

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md:6]:
```suggestion
The `BrowserType` is the entry point into launching a browser process; `chromium` is currently the only supported `BrowserType`. To use it, import `chromium` from the top level module `k6/x/browser`.
```

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md:14]:
```suggestion
| Property            | Type     | Default | Description                                                                                                                                                                                                                                                           |
```

File: src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md:58]:
```suggestion
| Property | Description                                                                                                                                                                                                                         |
```

File: src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md:66]: Do you think this is useful to document? `devices` is a map that users can inspect on their own, so maybe just the example should be enough. Besides, these are all quite old devices, which raises the question of when will modern devices be added. And once we start [updating the settings automatically](https://github.com/grafana/xk6-browser/issues/290), it would be a chore to keep the list here in sync.

So I'd vote for removing this section and just keeping the example.

File: src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md:97]: For illustrating a good practice :slightly_smiling_face::

```suggestion
  });
  
  page.close();
  browser.close();
```

File: src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md:60]:
```suggestion
| chromium   | A [BrowserType](/javascript-api/xk6-browser/browsertype) to launch tests in a Chromium-based browser.                                                                                                                                     |
```

File: src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md:24]:
```suggestion
The first step is to import the `chromium` [BrowserType](/javascript-api/xk6-browser/browsertype), and use its `launch` method to start up a Chromium [Browser](/javascript-api/xk6-browser/browser) process (which is currently the only available `BrowserType`). After it starts, you can interact with it using the [browser-level APIs](#browser-level-apis).
```

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md:10]: Do we want to document the [other methods](https://github.com/grafana/xk6-browser/blob/c5c2cead896d615f06f70f0f4ed24e1cd041b1ed/api/browser_type.go#L28-L34) as well?

`connect()` and `launchPersistentContext()` aren't implemented yet, but the others would be good to have. We should list the not-implemented ones here as well with a WIP icon, but we can leave out documenting their options, since they're not usable.

It would be nice if we could use the new collapsible element for documenting these options as well, so we don't have to use a separate table for it, though it probably wouldn't make sense, so nevermind. Let's just change the header below to "`launch()` options", so that it's clear. And then change this to:

```suggestion
| launch([options]) | Launches a new browser process. You can see a list of all possible `options` in [the table below](#launch-options). |
```

(Confirm that the link actually works... I'm not sure if our Gatsby JS will properly convert "`launch()` options" to `#launch-options`.)

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md:12]: To clarify:
```suggestion
### `launch()` options
```

We'll have to change it anyway once we implement `connect()` and `launchPersistentContext()`, which also take an `options` object.

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (imiric) [inline src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md:23]:
```suggestion
| proxy             | string   | `null`  | Specify to set browser's proxy configuration.                                                                                                                                                                                                                                |
```

**Review by imiric — CHANGES_REQUESTED** [review]

File: src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md
Comment (ankur22) [reply src/data/markdown/docs/30 xk6-browser/30 xk6-browser.md]: I was on the fence with this one too, so happy to remove it.

Resolved in: e9ced7265ed08fed9b7b203df46a3025e172824c

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (ankur22) [reply src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md]: Yeah, i forgot about those. I've added the missing methods. I've also moved the options for `launch` into its own page, just like we're doing with the other classes. This also means we can use the nested table for the options.

Resolved in: 805dd25465b382f0e5edc1797adad4ac3000d8a1

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (ankur22) [reply src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md]: I've made the larger change so that the options for `launch` now live in its own page. I've not created pages for `connect` or `laucnPersistentContext` since they're not implemented but documented them as WIP.

Resolved in: 805dd25465b382f0e5edc1797adad4ac3000d8a1

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (imiric) [reply src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md]: Nice, though I'm not sure if it makes sense to nest the options in that case. If they're on a separate page in their own table already, they can be shown expanded. We can use the collapsible element for nested sub-properties, though.

Comment (ankur22) [reply src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md]: I've removed the nesting for those options.

File: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md
Comment (imiric) [reply src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md:6]: Hey, I still see `/k6/x/browser` on b01db6f2.

Comment (ankur22) [reply src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md]: Hmm, it looks like i added it back in 🤔, i've removed it now in cbdbfaa2b28fffbab81284e72b58b93df165c449.

**Review by imiric — APPROVED** [review]
:+1:

Let's wait for Inanc's approval as well.

**Review by inancgumus — APPROVED** [review]
LGTM 👍

**Pull request merged**
